### PR TITLE
New arguments for specific patch

### PR DESCRIPTION
**Pull request by joemmetry**
Hi,

Personally, on my hackintosh, I only want to enable cpu renaming but still like to display the MacPro7,1 PCI and memory UIs via `-revnomemmgmtpatch`.

I thought that some might like only the PCI and memory UIs thus I support another checking for `-revnocpunamepatch`

Thanks

### CODE REVIEW

**Comment by vit9696** [comment]
Needs to have device property aliases.

**Comment by vit9696** [comment]
Or perhaps nvram, like in the other keys for this kext.

**Comment by vit9696** [comment]
Is this going to be fixed?

**Comment by joemmetry** [comment]
Sorry for the delay. Yes, just pushed an update.

File: RestrictEvents/RestrictEvents.cpp [hunk 1]
```diff
@@ -219,6 +222,37 @@ struct RestrictEventsPolicy {
...
+        
+		char *value = reinterpret_cast<char *>(&duip[0]);
+		value[sizeof(duip) - 1] = '\0';
+
+		if (strncmp(value, "all", 3) == 0) {
+			// Disable all UI patches
+			disableAllPatching = true;
+		} else if (strncmp(value, "mempci", 6) == 0) {
+			disableMemoryPciManagementPatching = true;
+			disableAllPatching = false;
```
Comment (vit9696) [inline RestrictEvents/RestrictEvents.cpp:244]: It is already false by being static, no need to explicitly assign it false.

File: RestrictEvents/RestrictEvents.cpp [hunk 2]
```diff
@@ -219,6 +222,37 @@ struct RestrictEventsPolicy {
...
+		if (PE_parse_boot_argn("revnopatch", duip, sizeof(duip))) {
+			DBGLOG("rev", "read revnopatch from boot-args");
+		} else if (readNvramVariable(NVRAM_PREFIX(LILU_VENDOR_GUID, "revnopatch"), u"revnopatch", &EfiRuntimeServices::LiluVendorGuid, duip, sizeof(duip))) {
+			DBGLOG("rev", "read revnopatch from NVRAM");
+		}
+        
+		char *value = reinterpret_cast<char *>(&duip[0]);
+		value[sizeof(duip) - 1] = '\0';
+
+		if (strncmp(value, "all", 3) == 0) {
```
Comment (vit9696) [inline RestrictEvents/RestrictEvents.cpp:239]: Use strcmp here and below.

File: RestrictEvents/RestrictEvents.cpp [hunk 3]
```diff
@@ -219,6 +222,37 @@ struct RestrictEventsPolicy {
 		return cc;
 	}
 
+	/**
+	 * Retrieve which system UI is to be disabled - all, mempci, cpuname, none (default)
+	 */
+	static void processDisableUIPatch() {
+    uint32_t duip[32] {};
```
Comment (vit9696) [inline RestrictEvents/RestrictEvents.cpp:229]: Indentation is off here.

Comment (vit9696) [reply RestrictEvents/RestrictEvents.cpp:229]: And it is probably better to use char duip[128].

File: RestrictEvents/RestrictEvents.cpp [hunk 4]
```diff
@@ -337,24 +371,27 @@ PluginConfiguration ADDPR(config) {
 	[]() {
 		DBGLOG("rev", "restriction policy plugin loaded");
 		verboseProcessLogging = checkKernelArgument("-revproc");
+    RestrictEventsPolicy::processDisableUIPatch();
```
Comment (vit9696) [inline RestrictEvents/RestrictEvents.cpp:374]: Indentation again.

**Pull request merged**